pymatgen/io/lobster/inputs.py

```diff
@@ -186,42 +180,35 @@ def diff(self, other):
         for k1, v1 in self.items():
             k1lower = k1.lower()
             if k1lower not in key_list_others:
-                different_param[k1.upper()] = {"lobsterin1": v1, "lobsterin2": None}
-            else:
-                for key_here in other:
-                    if k1.lower() == key_here.lower():
-                        new_key = key_here
-
-                if isinstance(v1, str):
-                    if v1.strip().lower() != other[new_key].strip().lower():
-                        different_param[k1.upper()] = {
-                            "lobsterin1": v1,
-                            "lobsterin2": other[new_key],
-                        }
-                    else:
-                        similar_param[k1.upper()] = v1
-                elif isinstance(v1, list):
-                    new_set1 = {element.strip().lower() for element in v1}
-                    new_set2 = {element.strip().lower() for element in other[new_key]}
-                    if new_set1 != new_set2:
-                        different_param[k1.upper()] = {
-                            "lobsterin1": v1,
-                            "lobsterin2": other[new_key],
-                        }
-                elif v1 != other[new_key]:
-                    different_param[k1.upper()] = {
+                different_param[k1.lower()] = {"lobsterin1": v1, "lobsterin2": None}
+            elif isinstance(v1, str):
+                if v1.strip().lower() != other[k1lower].strip().lower():
+                    different_param[k1.lower()] = {
                         "lobsterin1": v1,
-                        "lobsterin2": other[new_key],
+                        "lobsterin2": other[k1lower],
```

Comment:
Have to check this lart again. I think I overlooked something here.

Reply:
Fixed.

Reply:
Thanks for fixing! 

I'm currently getting the following `ruff` error:
```python
pymatgen/io/lobster/inputs.py:210:13: SIM102 Use a single `if` statement instead of nested `if` statements
```
from https://github.com/materialsproject/pymatgen/blob/aee9bc49b4fe95857b11d23eb279fb0a60d8bc60/pymatgen/io/lobster/inputs.py#L209-L213

I would correct it if it's not intended.

Meanwhile, I didn't try myself so I could be wrong, do we need to hard-code a `diff` method ourselves? Have you considered using the [symmetric_difference](https://docs.python.org/3/library/stdtypes.html#frozenset.symmetric_difference) to calculate the intersection and difference?

Reply:
It's again implemented to take care of cases.

I am not sure if this is fine. I would need to have a careful look.

Reply:
Yes, it's fine!

Reply:
> It's again implemented to take care of cases.

Yes I begin to understand your intention in `io.lobster` now (mostly to handle the case sensitivity issue).

> It looks fine on first sight.

I would experiment on this later and tell you how everything goes (maybe a bit later, I'm currently taking up too much work).

Thanks!